### PR TITLE
Update to support self hosted scms

### PR DESCRIPTION
**Pull request by krames**
In order to support self hosted scms, this pr updates github, bitbucket, and gitlab checkout methods to take a host. 

In the process of converting between external checkout options to the scm specific checkout options we check to see if an external host is passed in. If the host was not passed in we default to the cloud version of the provider.

### CODE REVIEW

**Comment by bfosberry** [comment]
👏

File: scm.go [hunk 1]
```diff
@@ -426,46 +441,66 @@ func convertExternalCheckoutOptions(externalCheckoutOptions *ExternalCheckoutOpt
...
 				Branch:            externalCheckoutOptions.Branch,
 				CommitID:          externalCheckoutOptions.CommitID,
 				CommitMessage:     externalCheckoutOptions.CommitMessage,
 				CommittedAt:       externalCheckoutOptions.CommittedAt,
 				SecurityOptions:   securityOptions,
-			}, nil
+			}
+
+			if opts.Host == "" {
+				opts.Host = "bitbucket.org"
```
Comment (GeorgeMac) [inline scm.go]: Might be worth making these addresses constants at the top 👍

Comment (krames) [reply scm.go]: From a memory point of view, does this recreate that string every time and the constant is only allocated once? Or is this comment more about having magic strings in the code?

Comment (markphelps) [reply scm.go]: `BITBUCKET_ORG`

Comment (krames) [reply scm.go]: I can't believe I missed that! Thanks!

**Pull request merged**